faf-commons-data/src/main/java/com/faforever/commons/replay/semantics/Semantics.java

```diff
@@ -0,0 +1,103 @@
+package com.faforever.commons.replay.semantics;
+
+import com.faforever.commons.replay.body.Event;
+import com.faforever.commons.replay.header.Source;
+import com.faforever.commons.replay.semantics.records.ChatMessage;
+import com.faforever.commons.replay.semantics.records.TrackedEvent;
+import com.faforever.commons.replay.shared.LuaTable;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+
+public class Semantics {
+
+  public static Duration tickToDuration(int tick) {
+    return Duration.ofSeconds(tick / 10);
+  }
+
+  public static List<TrackedEvent> registerEvents(List<Source> sources, List<Event> events) {
+    final int[] tick = {-1};
+    final int[] clientId = {-1};
```

Comment:
I would just use an AtomicInteger here

Reply:
Processed in https://github.com/FAForever/faf-java-commons/pull/143/commits/0f0aca220381953c2dadf6f3d5c80aeb03f5b722